client/main.go

```diff
@@ -467,6 +480,91 @@ func getNamespaceInfo(resourcePath, OSDFDirectorUrl string, isPut bool) (ns name
 	}
 }
 
+func schemeUnderstood(scheme string) error {
+	understoodSchemes := []string{"file", "osdf", "pelican", "stash", ""}
+
+	_, foundDest := Find(understoodSchemes, scheme)
+	if !foundDest {
+		return fmt.Errorf("Do not understand the destination scheme: %s. Permitted values are %s",
+			scheme, strings.Join(understoodSchemes, ", "))
+	}
+	return nil
+}
+
+func newPelicanURL(remoteUrl *url.URL, scheme string) (pelicanURL PelicanURL, err error) {
+	if remoteUrl.Host != "" {
+		if scheme == "osdf" || scheme == "stash" {
+			remoteUrl.Path, err = url.JoinPath(remoteUrl.Host, remoteUrl.Path)
```

Comment:
This section is used when someone does an osdf or stash url but only puts two `/`'s instead of 3 (hence they have a hostname). What would you like me to do differently? Create our own url like the comment below and modify that url?